src/test/kotlin/com/exactpro/th2/common/schema/filter/strategy/impl/TestAnyMessageFilterStrategy.kt

```diff
@@ -129,5 +337,163 @@ class TestAnyMessageFilterStrategy {
             arguments(RAW_MESSAGE_MATCH, true),
             arguments(RAW_MESSAGE_MISS_MATCH, false)
         ) + parsedMessages()
+
+        /////////////////
+
+        private val MATCHES_WITH_SAME_FIELDS = AnyMessage.newBuilder().setMessage(
+            message("test", Direction.FIRST, "test-alias").apply {
+                addField("test-field", "test-value1")
+                addField("test-field", "test-value2")
+            }
+        ).build()
+
+        @JvmStatic
+        fun messagesWithMultipleSameFields(): List<Arguments> = listOf(
+            arguments(MATCHES_WITH_SAME_FIELDS, true),
+        )
+
+        ////////////////
+
+        private val MESSAGE_WITH_ONE_FIELDS_FOR_FILTER_WITH_SAME_FIELDS = AnyMessage.newBuilder().setMessage(
+            message("test", Direction.FIRST, "test-alias").apply {
+                addField("test-field", "test-value1")
+            }
+        ).build()
+
+        @JvmStatic
+        fun messagesWithSameFilterFields(): List<Arguments> = listOf(
+            arguments(MESSAGE_WITH_ONE_FIELDS_FOR_FILTER_WITH_SAME_FIELDS, false),
+        )
+
+        @JvmStatic
+        fun messagesWithMultipleFiltersWithSameFilterField(): List<Arguments> = listOf(
+            arguments(MESSAGE_WITH_ONE_FIELDS_FOR_FILTER_WITH_SAME_FIELDS, true),
+        )
+
+        ///////////////
+
+        private val MULTIPLE_FILTERS_MESSAGE_FULL_MATCH = AnyMessage.newBuilder().setMessage(
```

Comment:
These contents are used only once, please make a method for building message and call it in static methods provided test parameters